scalarxmat44.c

```diff
@@ -0,0 +1,107 @@
+#include <immintrin.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/time.h>
+
+#define N 4
+
+int
+main() {
+
+    // allocate memory for array
+    int are_equal = 1;
+    float lamda = 5.0;
+    int count = 0;
+    float(*A)[N] = malloc(sizeof(float[N][N]));
+    if (!A) {
+        fprintf(stderr, "error allocating memory for matrix\n");
+        exit(1);
+    }
+    float(*B)[N] = malloc(sizeof(float[N][N]));
+    if (!B) {
+        fprintf(stderr, "error allocating memory for matrix\n");
+        exit(1);
+    }
+    // fill array with numbers
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            A[i][j] = (float) count++;
+        }
+    }
+    count = 0;
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            B[i][j] = (float) count++;
```

Comment:
I`d suggest B to have different kind of elements, eg. use a random or the reverse index, or count * constant.

Reply:
Ι used A to store the result of scalar * A4x4 and B4x4 to store the implementation of sse,avx2....
and then I compare the matrices

Reply:
oh sorry this is not for matrix multiplication, ignore